src/Util/ListUtil.js

```diff
@@ -17,3 +18,89 @@ export function slugifyListName(name) {
 
   return result;
 }
+
+/**
+ * Creates a new watchlist.
+ * @param {string} name The list name.
+ * @param {array} anime An array of animeIds.
+ * @param {boolean} privateList Is this a private list.
+ * @param {string} desc The users' description of the list.
+ * @param {object} user The user auth object.
+ * @param {object} localUser The localUser object.
+ * @param {string} malList MAL list name or NULL if a custom watchlist.
+ */
+export function createNewWatchlist(
+  name,
+  anime,
+  privateList,
+  desc,
+  user,
+  localUser,
+  malList
+) {
+  let listId = generateId();
+  let temp = { ...localUser };
+
+  // Build list object for a CUSTOM watchlist
+  if (!malList) {
+    temp.lists = [
+      ...(temp.lists ?? []),
+      {
+        name: name,
+        anime: anime ?? [],
+        privateList: privateList,
+        desc: desc,
+        id: listId,
+      },
+    ];
+  }
+  // Build list object for an imported MAL watchlist
+  else {
+    const syncDate = new Date();
+    temp.lists = [
+      ...(temp.lists ?? []),
+      {
+        name: getMalListName(malList),
+        anime: anime ?? [],
+        privateList: privateList,
+        desc: getMalListDesc(malList),
+        id: listId,
+        syncData: {
+          source: getMalListSource(malList),
+          syncDate,
+        },
+      },
+    ];
+  }
+
+  // Only registered users have their watchlistData created and displayed on '/home'
+  if (!privateList && !user.isAnonymous)
+    CreateWatchlistDataEntry(user.uid, listId);
```

Comment:
Do we want to make this an async function and await completion of the firestore operation here?

Reply:
Yes!